halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/providers/dcos/cluster/DCOSListClusterCommand.java

```diff
@@ -0,0 +1,56 @@
+package com.netflix.spinnaker.halyard.cli.command.v1.config.providers.dcos.cluster;
+
+import lombok.Getter;
+
+import com.beust.jcommander.Parameters;
+import com.netflix.spinnaker.halyard.cli.services.v1.Daemon;
+import com.netflix.spinnaker.halyard.cli.services.v1.OperationHandler;
+import com.netflix.spinnaker.halyard.cli.ui.v1.AnsiUi;
+import com.netflix.spinnaker.halyard.config.model.v1.node.Provider;
+import com.netflix.spinnaker.halyard.config.model.v1.providers.dcos.Cluster;
+import com.netflix.spinnaker.halyard.config.model.v1.providers.dcos.DCOSProvider;
+
+import java.util.List;
+
+/**
+ *
+ */
+@Parameters(separators = "=")
+public class DCOSListClusterCommand extends AbstractClusterCommand {
+  public String getDescription() {
+    return "List the cluster names for the " + getProviderName() + " provider.";
+  }
+
+  @Getter
+  private String commandName = "list";
+
+  private Provider getProvider() {
+    String currentDeployment = getCurrentDeployment();
+    String providerName = getProviderName();
+    return new OperationHandler<Provider>()
+        .setFailureMesssage("Failed to get provider " + providerName + ".")
+        .setOperation(Daemon.getProvider(currentDeployment, providerName, !noValidate))
+        .get();
+  }
+
+  @Override
+  protected void executeThis() {
+    Provider provider = getProvider();
+    if (!provider.providerType().equals(Provider.ProviderType.DCOS)) {
```

Comment:
Maybe my concern is that the exception below should instead indicate an assertion was violated, and isn't a user error.